### PR TITLE
12263 Fixed empty tax id bug

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#12263

*Description of changes:*
- deleted empty tax id so it doesn't fail schema pattern
- added unit tests
- fixed lint error
- app version = 4.0.7

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/mixins/filing-template-mixin.ts [hunk 1]
```diff
@@ -396,6 +396,10 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
         }
       }
       delete party.officer.businessNumber
+
+      // if tax id is empty (which doesn't match the schema pattern), delete it
+      if (!party.officer.taxId) delete party.officer.taxId
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:401]: If the user did not touch the input control for this value (which is labelled Business Number) then this v-model value is null or undefined, which is fine.

But if the user did touch the input control then this v-model value is "" (empty string), which fails schema validation due to the [required pattern].(https://github.com/bcgov/business-schemas/blob/e32e6dc561abebdc3034af48993da12f882dfb80/src/registry_schemas/schemas/party.json#L46).

Therefore, the solution is to delete this property if it's empty (null, undefined or empty string).

File: tests/unit/PaymentErrorDialog.spec.ts [hunk 2]
```diff
@@ -122,10 +122,11 @@ describe('Payment Error Dialog', () => {
...
         store,
-        propsData: { dialog: true, warnings: [
-          { message: 'Test Warning 1' },
-          { message: 'Test Warning 2' }
-        ]}
+        propsData: { dialog: true,
+          warnings: [
+            { message: 'Test Warning 1' },
+            { message: 'Test Warning 2' }
+          ] }
```
Comment (severinbeauvais) [inline tests/unit/PaymentErrorDialog.spec.ts:129]: Lint fixes only in this file.

NB: these are auto-fixed if you do `npm run lint`.

File: tests/unit/filing-template-mixin.spec.ts [hunk 3]
```diff
@@ -98,6 +97,69 @@ describe('Registration Filing', () => {
...
+      {
+        officer: {
+          id: '4321',
+          email: 'kitty@example.com',
+          lastName: '',
+          firstName: '',
+          partyType: 'organization',
+          middleName: '',
+          organizationName: 'Crazy Cat Consulting Company',
+          businessNumber: '123456789'
```
Comment (severinbeauvais) [inline tests/unit/filing-template-mixin.spec.ts:136]: Valid business number. It should be in the filing object.

File: tests/unit/filing-template-mixin.spec.ts [hunk 4]
```diff
@@ -144,7 +206,68 @@ describe('Registration Filing', () => {
...
+            {
+              officer: {
+                id: '1234',
+                email: 'lucille@bcregtest.gov.bc.ca',
+                lastName: 'TWENTY',
+                firstName: 'BCREGTEST Lucille',
+                partyType: 'person',
+                middleName: '',
+                organizationName: ''
+              },
```
Comment (severinbeauvais) [inline tests/unit/filing-template-mixin.spec.ts:220]: No tax id 👍

File: tests/unit/filing-template-mixin.spec.ts [hunk 5]
```diff
@@ -144,7 +206,68 @@ describe('Registration Filing', () => {
...
+            {
+              officer: {
+                id: '4321',
+                email: 'kitty@example.com',
+                lastName: '',
+                firstName: '',
+                partyType: 'organization',
+                middleName: '',
+                organizationName: 'Crazy Cat Consulting Company',
+                taxId: '123456789'
```
Comment (severinbeauvais) [inline tests/unit/filing-template-mixin.spec.ts:245]: Tax id 👍

**Review by lewischenstudio — APPROVED** [review]

**Pull request merged**